libs/x-content/src/test/java/org/elasticsearch/common/xcontent/ObjectParserTests.java

```diff
@@ -1058,20 +1061,42 @@ public void testCompatibleFieldDeclarations() throws IOException {
         {
             // new_name is allowed to be parsed with compatibility
             XContentParser parser = createParserWithCompatibilityFor(JsonXContent.jsonXContent, "{\"new_name\": 1}",
-                RestApiCompatibleVersion.minimumSupported());
+                minimumSupported());
             StructWithCompatibleFields o = StructWithCompatibleFields.PARSER.parse(parser, null);
             assertEquals(1, o.intField);
         }
         {
-
             // old_name is allowed to be parsed with compatibility, but results in deprecation
             XContentParser parser = createParserWithCompatibilityFor(JsonXContent.jsonXContent, "{\"old_name\": 1}",
-                RestApiCompatibleVersion.minimumSupported());
+                minimumSupported());
             StructWithCompatibleFields o = StructWithCompatibleFields.PARSER.parse(parser, null);
             assertEquals(1, o.intField);
             assertWarnings(false, "[struct_with_compatible_fields][1:14] " +
                 "Deprecated field [old_name] used, expected [new_name] instead");
 
         }
     }
+
+//    protected boolean enableWarningsCheck() {
```

Comment:
how do we test for future version? A class instead of enum would allow for Spying..

Reply:
I don't think you need to test the future version. Since it does not exist (yet) it can not be set. Once current == 9 it will get tested via the current(). However, i realize that is not the spirit of what you are trying to test here ... you are trying to test the "after" part of onOrAfter right ?

I think to test the spirit of what you want here is to declare a Parse field with `.onOrAfter(RestApiCompatibleVersion.minimumSupported())));` and a `createParserWithCompatibilityFor(..., currentVersion())` and ensure it still parses correctly.

Reply:
good idea, I created a test that uses minimumSupported ('pretending this being the version when a new field is introduced) and then I test that is being parsed for `current` version (which is the "future" version)